scripts/irods/test/test_ifsck.py

```diff
@@ -83,3 +83,31 @@ def test_ifsck_continues_when_user_cannot_access_subdirectory__issue_5358(self):
             for d in dirs:
                 os.chmod(d, 0o777)
 
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing, checks vault")
+    def test_ifsck_unregistered_file_detected__6467(self):
+        directory_name = 'test_ifsck_unregistered_file_detected__6467'
+        filename1 = 'test_ifsck_unregistered_file_detected__6467_f1'
+        filename2 = 'test_ifsck_unregistered_file_detected__6467_f2'
+        full_directory_path = '%s/%s' % (os.getcwd(), directory_name)
+        full_directory_logical_path = '%s/%s' % (self.admin.session_collection, directory_name)
+        full_path_filename1 = '%s/%s' % (full_directory_path, filename1)
+        full_path_filename2 = '%s/%s' % (full_directory_path, filename2)
+        try:
+            # create directory with file and register it
+            os.mkdir(full_directory_path)
+            filesize = 100
+            lib.make_file(full_path_filename1, filesize)
+            self.admin.assert_icommand(['ireg', '-C', full_directory_path, full_directory_logical_path])
+            # verify ifsck returns no error
+            self.admin.assert_icommand(['ifsck', '-r', full_directory_path])
+            # create a new unregisterd file
+            lib.make_file(full_path_filename2, filesize)
+            # verify ifsck errors
+            _, _, rc = self.admin.assert_icommand(['ifsck', '-r', full_directory_path], 'STDOUT_SINGLELINE', ['WARNING', filename2, 'is not registered in iRODS'])
```

Comment:
Consider changing the expected output to be the exact string you're looking for.
```
'WARNING: local file [{0}] is not registered in iRODS.'.format(filename2)
```

As of right now, it's possible that a tweak to the icommand could lead to false positives.

Reply:
done